Updates to enable CI via VSTS public projects

d3r3kk opened the pull request:
Small tweaks that enable VSTS builds against this repo.

- Update `tests` module to support the output of VSTS-compatible file formats for JUnit and Cobertura
- Create `flake8` specific config file - one for local use and one for ci
- Switch the build badge in the readme.md file to the VSTS badge

**NOTE**: Don't merge this to the Microsoft/ptvsd repo until we are ready to make the full switch to VSTS.

d3r3kk committed:
Add flake8 configuration files

- .flake8: main one (a default file picked up by flake) for dev use

- .flake8.ci: addendum config for CI to make use of only.

- this way we can allow our contributors to simply run > python -m flake8 on the command line, or even more simply > flake8

d3r3kk committed:
Update debugProtocol.json to latest

d3r3kk committed:
Set build badge to the CI in VSTS

d3r3kk committed:
Enable JUnit style test file output.
- add arg to specify the output location of a JUnit-compatible log file
- defaults to current folder with a descriptive name

d3r3kk committed:
Correct linter warnings

d3r3kk committed:
Update tests of the tests module
- add test for setting junit report file
- add missing return param for junit file to  all tests
- add trivial assert to use junit file return value

d3r3kk commented:
@ericsnowcurrently, @zooba, @DonJayamanne, have a look when you have a chance and let me know what could be made better...

d3r3kk committed:
PR suggestions

- keep travis badge
- pep8 infractions/fixups
- convert_args return value simplified
- allow original style of unittest reporting as well
- remove unnecessary flags from junit xmlrunner

zooba commented on tests/test_tests___main__.py:
```diff
@@ -10,127 +10,161 @@
 
-        self.assertEqual(argv, [
+        self.assertEqual(runtest_args.argv, [
             sys.executable + ' -m unittest',
```
Not sure where this is coming from, but the `-m` and `unittest` parts should be separate list items (assuming this command will be run at some point, at which point quoting rules may corrupt the first item here)

ericsnowcurrently replied:
In this case the code is correct.  Unittest is quirky in a few spots like this.

d3r3kk committed:
Use argparse to collect runtime configuration of tests module

- change as little as possible to incorporate this change for now

d3r3kk committed:
Add back a missing argument

- argument --full was missing from the args to parse

d3r3kk committed:
Correct usage of multiple args affecting the same var

- set default to non-quick tests

d3r3kk committed:
Correct tests such that coverage completes successfully.

- Removing the args list broke tests in coverage scenarios
- Specific tests introduced  must be skipped if no network is enabled
- correct tests on the tests module

d3r3kk committed:
Fix linter problems

d3r3kk commented:
Closed as Microsoft/ptvsd#431 is where the real build will end up!
